janis_core/translations/cwl.py

```diff
@@ -1305,21 +1319,68 @@ def get_run_ref_from_subtool(
             )
 
 
+def add_when_conditional_for_workflow_stp(stp: cwlgen.WorkflowStep, when: Selector):
+    prepare_alias = lambda x: f"__when_{re.sub('[^0-9a-zA-Z]+', '', x)}"
+
+    # two step process
+    #   1. Look through and find ALL sources includng an operator's leaves
+    #           Ensure these are connected using the alias
+    #   2. Go through sources again, build up the expression
+
+    param_aliasing = {}
+    # Use a dict to ensure we don't double add inputs
+    ins_to_connect: Dict[str, cwlgen.WorkflowStepInput] = {}
+
+    processed_sources = []
+
+    src = when
+    if isinstance(src, InputNodeSelector) and isinstance(
+        src.input_node.default, Selector
+    ):
+        src = If(IsDefined(src), src, src.input_node.default)
+
+    processed_sources.append(src)
+
+    if isinstance(src, Operator):
+        # we'll need to get the leaves and do extra mappings
+        for leaf in src.get_leaves():
+            if not isinstance(leaf, Selector):
+                # probably a python literal
+                continue
+            sel = CwlTranslator.unwrap_selector_for_reference(leaf)
+            alias = prepare_alias(sel)
+            param_aliasing[sel] = alias
+            ins_to_connect[alias] = cwlgen.WorkflowStepInput(id=alias, source=sel)
+    else:
+        sel = CwlTranslator.unwrap_selector_for_reference(src)
+        alias = prepare_alias(sel)
+        param_aliasing[sel] = alias
+        ins_to_connect[alias] = cwlgen.WorkflowStepInput(id=alias, source=sel)
```

Comment:
Just thinking we may be able to avoid repeated code here (for easier maintenance). Something like:
```
nodes = [src]
if isinstance(src, Operator):
   nodes = src.get_leaves()
``` 
And we can apply the same code to the `nodes` list (so no need for the `else`.